### PR TITLE
Make README and Module docstring examples runnable

### PR DESCRIPTION
**Pull request by n2cholas**
This includes fixing minor typos, imports, code initialize variables + a forward pass.

## Checklist
- [x] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [x] This change is discussed in #1325 (doesn't close this issue, though)
- [x] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/master/docs/README.md#how-to-write-code-documentation).
- [ ] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

### CODE REVIEW

File: flax/linen/module.py [hunk 1]
```diff
@@ -1037,21 +1046,29 @@ def sow(self, col: str, name: str, value: T,
...
+
       class Foo(nn.Module):
         @nn.compact
         def __call__(self, x):
           h = nn.Dense(4)(x)
           self.sow('intermediates', 'h', h)
           return nn.Dense(2)(h)
-      y, state = Foo.apply(params, x, mutable=['intermediates'])
+
+      x = jnp.ones(((16, 9)))
```
Comment (andsteing) [inline flax/linen/module.py:1060]: One parens too many?

File: flax/linen/module.py [hunk 2]
```diff
@@ -1061,8 +1078,11 @@ def __call__(self, x):
           self.sow('intermediates', 'h', x * 2,
                    init_fn=init_fn, reduce_fn=reduce_fn)
           return x
-      y, state = Foo.apply(params, 1, mutable=['intermediates'])
-      print(state['intermediates'])  # ==> {'h': 3}
+
+      model = Foo2()
+      variables, _ = model.init(jax.random.PRNGKey(0), x).pop('intermediates')
```
Comment (andsteing) [inline flax/linen/module.py:1083]: (note that `variables` is actually empty here...)

File: flax/linen/module.py [hunk 3]
```diff
@@ -1037,21 +1046,29 @@ def sow(self, col: str, name: str, value: T,
...
         @nn.compact
         def __call__(self, x):
           h = nn.Dense(4)(x)
           self.sow('intermediates', 'h', h)
           return nn.Dense(2)(h)
-      y, state = Foo.apply(params, x, mutable=['intermediates'])
+
+      x = jnp.ones(((16, 9)))
+      model = Foo()
+      variables = model.init(jax.random.PRNGKey(0), x)
```
Comment (andsteing) [inline flax/linen/module.py:1062]: Why not the same as below?

```
variables, _ = model.init(jax.random.PRNGKey(0), x).pop('intermediates')
```

**Review by andsteing — CHANGES_REQUESTED** [review]
Thanks for the update!

Some small suggestions...

File: flax/linen/module.py (hunk 1 above)
Comment (n2cholas) [reply flax/linen/module.py]: Thanks for catching this, fixed!

File: flax/linen/module.py (hunk 2 above)
Comment (n2cholas) [reply flax/linen/module.py]: Thanks, fixed by making it the same as the previous example.

File: flax/linen/module.py (hunk 3 above)
Comment (n2cholas) [reply flax/linen/module.py:1062]: Given #1284, `intermediates` will no longer be in the collection after init. I mistakenly updated the top example and missed the bottom one. Now both are the same.

**Comment by n2cholas** [comment]
The build is failing with `Exception: ROM is missing for beam_rider` even though I rebased with master (which includes a fix to include that dependency.

**Comment by andsteing** [comment]
wrt build error : not sure why you still see that; can you try rebasing & rebuilding again?
Otherwise maybe @jheek knows what's going on with the `ppo` example failing the build...

**Review by andsteing — APPROVED** [review]

**Comment by n2cholas** [comment]
Done, but the build failed again with the same issue. The commit I rebased on (36fccf2) failed in the same way: https://github.com/google/flax/runs/2655565828

**Comment by jheek** [comment]
Sorry @n2cholas my original fix turned out not to work. This time rebasing should really fix the issue

**Comment by n2cholas** [comment]
Thanks @jheek, looks like third time's the charm!

**Pull request merged**